src/engine/controls/loopingcontrol.cpp

```diff
@@ -199,6 +199,12 @@ LoopingControl::LoopingControl(const QString& group,
             this, &LoopingControl::slotLoopDouble);
 
     m_pPlayButton = ControlObject::getControl(ConfigKey(group, "play"));
+
+    // Set NoAssertIfMissing only for ReadAheadManagerTest.FractionalFrameLoop
+    // where the "repeat" is not created (created by EngineBuffer which owns
+    // LoopingControl)
+    m_pRepeat = make_parented<ControlProxy>(
+            ConfigKey(group, "repeat"), this, ControlFlag::NoAssertIfMissing);
```

Comment:
Yes, this may be considered a hack to not bother the tests, and `repeat` _might_ break unnoticed if the CO is not created elsewhere (i.e. potentially not available here because it's created later).
Though, currently there is no test for repeat either.

Reply:
okay, after looking at how test mock certain engine controls I figured how to do it properly: like `quantize`,`repeat` is now also created in ReadAheadManagerTest and the assertion for the actual control is still in place.